libbeat/testing/integration/output_watcher_test.go

```diff
@@ -85,6 +87,90 @@ func TestOutputWatcher(t *testing.T) {
 		}
 	})
 
+	t.Run("JSON watcher", func(t *testing.T) {
+
+		InputDoc := mapstr.M{
+			"message": "sample dummy message",
+			"fields": mapstr.M{
+				"number": 20,
+				"slice":  []string{"20", "30", "40"},
+			},
+			"isthisboolean": false,
+			"nilValue":      nil,
+		}
+
+		cases := []struct {
+			name           string
+			fieldsToMatch  mapstr.M
+			expectObserved bool
+		}{
+			{
+				name: "test when expected value does not exist",
+				fieldsToMatch: mapstr.M{
+					"message": "sample message",
+				},
+				expectObserved: false,
+			},
+			{
+				name: "test when expected value is string type",
+				fieldsToMatch: mapstr.M{
+					"message": "sample dummy message",
+				},
+				expectObserved: true,
+			},
+			{
+				name: "test when expected value is int type",
+				fieldsToMatch: mapstr.M{
+					"fields": mapstr.M{
+						"number": float64(20),
+					},
+				},
+				expectObserved: true,
+			},
+			{
+				name: "test when expected value is slice type",
+				fieldsToMatch: mapstr.M{
+					"fields": mapstr.M{
+						"slice": []interface{}{"20", "30", "40"},
+					},
+				},
+				expectObserved: true,
+			},
+			{
+				name: "test when expected value is boolean type",
+				fieldsToMatch: mapstr.M{
+					"isthisboolean": false,
+				},
+				expectObserved: true,
+			},
+			{
+				name: "test when expected value is nil type",
+				fieldsToMatch: mapstr.M{
+					"nilValue": nil,
+				},
+				expectObserved: true,
+			},
+			{
+				name: "test when expected value is submap type",
+				fieldsToMatch: mapstr.M{
+					"fields": mapstr.M{
+						"number": float64(20),
+						"slice":  []interface{}{"20", "30", "40"},
+					},
+				},
+				expectObserved: true,
+			},
+		}
+
+		for _, tc := range cases {
+			t.Run(tc.name, func(t *testing.T) {
+				w := NewJSONWatcher(tc.fieldsToMatch)
+				w.Inspect(InputDoc.String())
```

Comment:
Basically this should be just `InputDoc` as a raw JSON string, without type conversions back in forth.